sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -101,6 +103,17 @@ private static void CompareActualToExpected(IEnumerable<Diagnostic> diagnostics,
             }
         }
 
+        private static void DumpActualDiagnostics(IEnumerable<Diagnostic> diagnostics)
+        {
+            Console.WriteLine($"Actual diagnostics: {diagnostics.Count()}");
+            foreach(var d in diagnostics.OrderBy(x => x.GetLineNumberToReport()))
+            {
+                var lineSpan = d.Location.GetLineSpan();
+                Console.WriteLine($"  Id: {d.Id}, Line: {d.Location.GetLineNumberToReport()}, " +
+                    $"[{lineSpan.StartLinePosition.Character}, {lineSpan.EndLinePosition.Character}]");
+            }
+        }
+
```

Comment:
Miscellaneous: helps diagnose test failures. The test result pane has a link to the _Output_ window, where this output will appear.